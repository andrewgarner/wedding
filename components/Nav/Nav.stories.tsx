import { Meta, Story } from "@storybook/react";
import React from "react";
import { Nav } from "./Nav";

export default {
  title: "Layout/Nav",
  component: Nav,
} as Meta;

const Template: Story = (args) => <Nav />;

export const Default = Template.bind({});
