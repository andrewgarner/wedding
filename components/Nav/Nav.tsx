import Link from "next/link";
import { useRouter } from "next/router";
import React, { VoidFunctionComponent } from "react";
import { Text, Typography } from "../Text";

const Nav: VoidFunctionComponent = () => {
  const router = useRouter();
  return (
    <nav className="">
      <ul className="inline-flex flex-row uppercase space-x-4">
        {router?.pathname !== "/" && (
          <Text tag="li" typography={Typography.Small}>
            <Link href="/" passHref>
              <a>Home</a>
            </Link>
          </Text>
        )}
        <Text tag="li" typography={Typography.Small}>
          <Link href="/wedding" passHref>
            <a>Wedding</a>
          </Link>
        </Text>
        <Text tag="li" typography={Typography.Small}>
          <Link href="/travel-and-stay" passHref>
            <a>Travel & stay</a>
          </Link>
        </Text>
        <Text tag="li" typography={Typography.Small}>
          <Link href="/rsvp" passHref>
            <a>RSVP</a>
          </Link>
        </Text>
      </ul>
    </nav>
  );
};

export { Nav };
