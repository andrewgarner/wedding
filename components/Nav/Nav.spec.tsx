import { render } from "@testing-library/react";
import React from "react";
import { Nav } from "./Nav";

describe("The navigation component", () => {
  it("renders correctly", () => {
    const { container } = render(<Nav />);

    expect(container).toMatchSnapshot();
  });
});
