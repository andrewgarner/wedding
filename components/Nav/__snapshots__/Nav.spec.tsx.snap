// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The navigation component renders correctly 1`] = `
<div>
  <nav
    class=""
  >
    <ul
      class="inline-flex flex-row uppercase space-x-4"
    >
      <li
        class="mx-auto max-w-prose font-sans text-sm lg:text-base font-normal"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li
        class="mx-auto max-w-prose font-sans text-sm lg:text-base font-normal"
      >
        <a
          href="/wedding"
        >
          Wedding
        </a>
      </li>
      <li
        class="mx-auto max-w-prose font-sans text-sm lg:text-base font-normal"
      >
        <a
          href="/travel-and-stay"
        >
          Travel & stay
        </a>
      </li>
      <li
        class="mx-auto max-w-prose font-sans text-sm lg:text-base font-normal"
      >
        <a
          href="https://prezola.com/wishlists/10241516"
        >
          Gift list
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
